### PR TITLE
Check if quic is enabled before warming up quic connections

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
QUIC warmup service tries to warmup connection even though QUIC is not enabled

#### Summary of Changes
Warmup QUIC connections only if QUIC has been enabled.

Fixes #

### CODE REVIEW

File: core/src/warm_quic_cache_service.rs [hunk 1]
```diff
@@ -36,23 +36,25 @@ impl WarmQuicCacheService {
...
-                        .unwrap()
-                        .leader_after_n_slots((CACHE_OFFSET_SLOT + slot_jitter) as u64);
-                    if let Some(leader_pubkey) = leader_pubkey {
-                        if maybe_last_leader
-                            .map_or(true, |last_leader| last_leader != leader_pubkey)
-                        {
-                            maybe_last_leader = Some(leader_pubkey);
-                            if let Some(addr) = cluster_info
-                                .lookup_contact_info(&leader_pubkey, |leader| leader.tpu)
+                    if use_quic() {
```
Comment (t-nelson) [inline core/src/warm_quic_cache_service.rs:39]: can we just not start the service instead?

Comment (pgarg66) [reply core/src/warm_quic_cache_service.rs:39]: Tried doing that initially. That meant plumbing `use_quic` flag from `validator/src/main.rs` to `validator.rs` to `tvu.rs`. Also, this change enabled somewhat dynamic control over when `quic` actually gets enabled.

Comment (pgarg66) [reply core/src/warm_quic_cache_service.rs]: Updated the PR to not start the warmup service if quic is not enabled.

**Review by sakridge — APPROVED** [review]
lgtm

**Pull request merged**